File: 0696-count-binary-substrings/0696-count-binary-substrings.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        n = len(s)
        one , zero = 0 , 0
        l, r = 0, 0
        ans = 0
        while r < len(s):
            while r < n and s[r] == "0":
                zero += 1
                r += 1
            if zero and one:
                ans += min(zero, one)
            one = 0
            while  r < n and s[r] == "1":
                one += 1
                r += 1
            
            if zero and one:
                ans += min(zero, one)
            zero = 0
        return ans

File: 0696-count-binary-substrings/0696-count-binary-substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
        int ans = 0;
        int one = 0, zero = 0;
        int n = s.size();
        int r = 0;
        while (r < n){
            while(r < n && s[r] == '0'){
                zero++;
                r++;
            }
            if (zero && one){
                ans += min(zero, one);
            }
            one = 0;
            while(r < n && s[r] == '1'){
                one++;
                r++;
            }
            if (zero && one){
                ans += min(zero, one);
            }

            zero = 0;

        }
        return ans;
    }
};